### DISCUSSION TITLE
Scale variation grids

### DISCUSSION THREAD

**Discussion by cschwan** [NNPDF/pineappl#163] [Ideas]
According to the [aMCfast](https://arxiv.org/abs/1406.7693) paper (first paragraph in section 2.4) it's possible to get the scale-varied results even without the renormalization/factorization-scale grids; they refer to the [APPLgrid paper](https://arxiv.org/abs/0911.2985), specifically eq. (14).

Do we want to want to implement this using PineAPPL + EKO? APPLgrid has the same feature, but using HOPPET.

**Comment by alecandido** [comment]
In principle it should be possible: in the case of DIS we are actually [doing it explicitly in yadism](https://github.com/N3PDF/yadism/blob/ede3ea6d5d47de6fcf45a5058c2704f2ca656922/src/yadism/esf/esf.py#L155-L160).

`ker_orders` are essentially the separate contributions that we use [to fill a PineAPPL grid](https://github.com/N3PDF/yadism/blob/ede3ea6d5d47de6fcf45a5058c2704f2ca656922/src/yadism/output.py#L234-L253) (indeed they are later assigned to [`obs.orders`](https://github.com/N3PDF/yadism/blob/ede3ea6d5d47de6fcf45a5058c2704f2ca656922/src/yadism/esf/esf.py#L171-L173), the one consumed to actually fill the PineAPPL grid).

I guess we can do the exact same for hadronic events, we just need the actual formula, the counter-parts of this one:
https://yadism.readthedocs.io/en/latest/theory/scale-variations.html#factorization
and this one:
https://yadism.readthedocs.io/en/latest/theory/scale-variations.html#renormalization

We always need the coefficients of the respective RGE: splitting functions for DGLAP (fact scale), and beta for alphas running (ren scale).
Both are available in EKO.

**Reply by alecandido** [reply]
Of course we'll need to do two different things for hadronic and DIS (in general for processes with 1 and 2 protons, we do not depend on the specific process).

DIS-like should be strictly simpler, but given that is already available in `yadism`, I'd do it immediately only if we need a benchmark. Otherwise I would go directly for hadronic, that will be the practical use case.

**Reply by alecandido** [reply]
Ah, wait a second, maybe we'll need actually yadism: in EKO splitting functions are only present in N-space, but we'll need them in x-space.
